app/Controllers/Http/CuratedTweetsController.ts

```diff
@@ -0,0 +1,86 @@
+import type { HttpContextContract } from "@ioc:Adonis/Core/HttpContext";
+import BaseController from "./BaseController";
+import TwitterPublic from "App/Services/TwitterPublic";
+import CuratedTweet from "App/Models/CuratedTweet";
+import { DateTime } from "luxon";
+
+export default class CuratedTweetsController extends BaseController {
+  public async store({ request, response }: HttpContextContract) {
+    const { ids } = request.only(["ids"]);
+    if (!Array.isArray(ids) || !ids.length) {
+      return response.status(400).json({ error: "No tweet IDs provided" });
+    }
+
+    const twitterClient = new TwitterPublic();
+
+    const twitterResponse = await twitterClient.getTweetsByIds(ids);
+
+    if (!twitterResponse || !twitterResponse.data?.length) {
+      return response.status(404).json({ error: "No data returned from Twitter" });
+    }
+
+    const usersById: Record<string, any> = {};
+    if (twitterResponse.includes?.users) {
+      for (const user of twitterResponse.includes.users) {
+        usersById[user.id] = user;
+      }
+    }
+
+    const mediaByKey: Record<string, any> = {};
+    if (twitterResponse.includes?.media) {
+      for (const m of twitterResponse.includes.media) {
+        mediaByKey[m.media_key] = m;
+      }
+    }
+
+    const results: CuratedTweet[] = [];
+    for (const tweet of twitterResponse.data) {
+      const user = tweet.author_id ? usersById[tweet.author_id] : null;
+      let mediaUrl: string | null = null;
+
+      if (tweet.attachments?.media_keys?.length) {
+        const firstKey = tweet.attachments.media_keys[0];
+        const media = mediaByKey[firstKey];
+        if (media) {
+          mediaUrl = media.url || media.preview_image_url || null;
+        }
+      }
+
+      const curatedTweet = await CuratedTweet.firstOrCreate(
```

Comment:
All the business logic of fetching the tweet belongs in the service imo

Controllers should be as dumb as possible (just the glue to tie HTTP to business logic)